tests/garage/tf/models/test_new_model_pickle.py

```diff
@@ -0,0 +1,63 @@
+import pickle
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.models.gaussian_mlp_model import GaussianMLPModel
+
+
+class TestNewModelPickling(unittest.TestCase):
+    def test_model_pickle(self):
+        sess = tf.Session()
+        with sess:
+            input_var = tf.placeholder(tf.float32, shape=(None, 5))
+            model = GaussianMLPModel(output_dim=2)
+            model.build(input_var)
+            data = np.random.random((3, 5))
+            results = sess.run(model.outputs[:2], feed_dict={input_var: data})
+            model_data = pickle.dumps(model)
+
+        sess.close()
+        tf.reset_default_graph()
```

Comment:
1. Why reset default graph here?
2. I think this op is strange here because you don't know which graph is default and where it is created. Why don't you create the sess inside a graph?
with tf.Graph().as_default() something like this

Reply:
I want to remove all previous defined variables to illustrate pickling.

Reply:
Thanks for the explanation. I replied this issue in another comment.